app/src/components/templateEditor/templateEditorUtils.ts

```diff
@@ -0,0 +1,46 @@
+import { assertUnreachable } from "@phoenix/typeUtils";
+
+import { extractVariablesFromFString, formatFString } from "./language/fString";
+import {
+  extractVariablesFromMustacheLike,
+  formatMustacheLike,
+} from "./language/mustacheLike";
+import { TemplateLanguages } from "./constants";
+import { TemplateLanguage } from "./types";
+
+/**
+ * Get an object of isomorphic functions for processing templates of the given language
+ *
+ * @param templateLanguage - The language of the template to process
+ *
+ * @returns An object containing the `format` and `extractVariables` functions.
+ * These functions share the same signature despite the different underlying
+ * templating languages.
+ */
+export const getTemplateLanguageUtils = (
+  templateLanguage: TemplateLanguage
+): {
+  format: ({
+    text,
+    variables,
+  }: {
+    text: string;
+    variables: Record<string, string | number | boolean | undefined>;
+  }) => string;
+  extractVariables: (template: string) => string[];
```

Comment:
might pull these out into some top level types instead of inline
eg:
```typescript
type ExtractVariablesFn = (template: string) => string[];
```